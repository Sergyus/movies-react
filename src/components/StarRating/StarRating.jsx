import React from 'react';
import PropTypes from 'prop-types';
import Star from './components/Star';

const StarRating = (props) => {
  const {
    starsSelected,
  } = props;
  return (
    <>
      {[...Array(Math.round(starsSelected))].map((n, i) => (
        <Star
          // eslint-disable-next-line react/no-array-index-key
          key={i}
        />
      ))}
    </>
  );
};

StarRating.propTypes = {
  starsSelected: PropTypes.number,
};

StarRating.defaultProps = {
  starsSelected: null,
};

export default StarRating;
