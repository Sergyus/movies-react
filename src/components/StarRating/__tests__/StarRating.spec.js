import React from 'react';
import { create } from 'react-test-renderer';
import StarRating from '..';

describe('StarRating component renders correctly', () => {
  it('StarRating renders correctly', () => {
    const tree = create(<StarRating starsSelected={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
