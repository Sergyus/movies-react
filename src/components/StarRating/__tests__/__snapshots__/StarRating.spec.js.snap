// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating component renders correctly StarRating renders correctly 1`] = `
Array [
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
]
`;
